internal/fs/fs.go

```diff
@@ -219,6 +219,16 @@ func NewFileSystem(
 }
 
 func createFileCacheHandler(cfg *ServerConfig) (fileCacheHandler *file.CacheHandler, err error) {
+	// If parallel downloads are enabled then download-parallelism-per-file or
+	// max-download-parallelism or read-request-size-mb can't be 0.
+	if cfg.MountConfig.FileCacheConfig.EnableParallelDownloads {
```

Comment:
How about doing this validation as part of config-parsing i.e. in func (fileCacheConfig *FileCacheConfig) validate() error

Reply:
I kept it that way because i want to keep yaml_parser to just have simple checks on the input values and here it required little business logic.
Now, i have made this check simple inside yaml_parser only and removed this if condition.

Reply:
I think we should resolve configuration related validations at one place unless it's impossible: for instance, if we keep simple configuration in the YAML, and more complicated logic here, there is fragmentation of validation logic and it's not straightforward to understand what the validation associated with a field is. And it would also impact readability. Maybe I'm missing something; something that we can discuss offline.

Reply:
Sure, we can discuss offline and fix this separately if need be.